### PR TITLE
home-assistant: 0.115.6 -> 0.116.2; python3Packages.voluptuous: 0.11.7 -> 0.12.0; pythonPackages.yeelight: 0.5.3 -> 0.5.4

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

New upstream release. Bumped voluptuous, since they required a newer version released two weeks ago.

Also enabled parallel testing using xdist, which cuts down the testPhase by a large margin.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
@test home-assistant.

File: pkgs/servers/home-assistant/default.nix [hunk 1]
```diff
@@ -88,50 +81,80 @@ in with py.pkgs; buildPythonApplication rec {
...
+          and not test_merge_id_schema \
+          and not test_merge \
+          and not test_create_default_config \
+          and not test_ensure_config_exists_creates_config \
+          and not test_ensure_config_exists_uses_existing_config \
+          and not test_load_yaml_config_converts_empty_files_to_dict \
+          and not test_load_yaml_config_preserves_key_order \
+          and not test_load_yaml_config_raises_error_if_malformed_yaml \
+          and not test_load_yaml_config_raises_error_if_not_dict \
+          and not test_load_yaml_config_raises_error_if_unsafe_yaml"
```
Comment (jonringer) [inline pkgs/servers/home-assistant/default.nix:146]: using pytestCheckHook would make this look a less intimidating, not to mention you can use comments more effectively
```suggestion
  disabledTests = [
          "test_load_yaml_config_converts_empty_files_to_dict"
          "test_load_yaml_config_preserves_key_order"
          "test_load_yaml_config_raises_error_if_malformed_yaml"
          "test_load_yaml_config_raises_error_if_not_dict"
          "test_load_yaml_config_raises_error_if_unsafe_yaml"
        ];
```

**Review by jonringer — APPROVED** [review]
diff LGTM
if you don't want to rewrite the test logic, then thats all good :)

```
https://github.com/NixOS/nixpkgs/pull/99986
27 packages built:
appdaemon elasticsearch-curator esphome home-assistant python27Packages.voluptuous python37Packages.async-upnp-client python37Packages.bellows python37Packages.pyhs100 python37Packages.voluptuous python37Packages.voluptuous-serialize python37Packages.zha-quirks python37Packages.zigpy python37Packages.zigpy-cc python37Packages.zigpy-deconz python37Packages.zigpy-xbee python37Packages.zigpy-zigate python38Packages.async-upnp-client python38Packages.bellows python38Packages.pyhs100 python38Packages.voluptuous python38Packages.voluptuous-serialize python38Packages.zha-quirks python38Packages.zigpy python38Packages.zigpy-cc python38Packages.zigpy-deconz python38Packages.zigpy-xbee python38Packages.zigpy-zigate
```

File: pkgs/servers/home-assistant/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix:146]: I'm all for using the hook, I wonder why this was split into two pytest runs though.

The first one excludes `test/components/`, the second runs explicit components from with `test/components/`. Is that possible within a single pytest run?

Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix:146]: Turns out: Yes, it is. WIP.

Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix]: Done. Though, I fear I need to do some more test runs to iron out the list racy tests.

**Comment by mweinelt** [comment]
Meanwhile: https://github.com/home-assistant/core/releases/tag/0.116.1

File: pkgs/servers/home-assistant/default.nix [hunk 2]
```diff
@@ -88,50 +83,108 @@ in with py.pkgs; buildPythonApplication rec {
...
     repo = "core";
     rev = version;
-    sha256 = "07j54glcpa5ngkr0pwdg44f8gas3jz3nh653mr5sb5wg7xspgcr8";
+    sha256 = "182iw9dbz15zj47npvpxp531q3wdw81qnsy663d2bkvqlqszbiw5";
   };
 
+  patches = [
+    (fetchpatch {
+      #  Fix group tests when run in parallel, remove >= 0.117.0
+      url = "https://github.com/home-assistant/core/pull/41446.patch";
```
Comment (jonringer) [inline pkgs/servers/home-assistant/default.nix:92]: please use a stable uri (could be force pushed to)
```suggestion
      url = "https://github.com/home-assistant/core/commit/896f64d02b311b313fe04bff366be452381be49d.patch";
```

Comment (mweinelt) [reply pkgs/servers/home-assistant/default.nix]: Indeed, thanks for pointing that out.

**Comment by mweinelt** [comment]
And https://github.com/home-assistant/core/releases/tag/0.116.2.

I'll merge this after @GrahamcOfBorg confirms all well, else we're gonna stack more updates. They're coming fast these days.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 99986` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>appdaemon</li>
    <li>elasticsearch-curator</li>
    <li>esphome</li>
    <li>home-assistant</li>
    <li>python27Packages.voluptuous</li>
    <li>python27Packages.yeelight</li>
    <li>python37Packages.async-upnp-client</li>
    <li>python37Packages.bellows</li>
    <li>python37Packages.pyhs100</li>
    <li>python37Packages.voluptuous</li>
    <li>python37Packages.voluptuous-serialize</li>
    <li>python37Packages.yeelight</li>
    <li>python37Packages.zha-quirks</li>
    <li>python37Packages.zigpy</li>
    <li>python37Packages.zigpy-cc</li>
    <li>python37Packages.zigpy-deconz</li>
    <li>python37Packages.zigpy-xbee</li>
    <li>python37Packages.zigpy-zigate</li>
    <li>python38Packages.async-upnp-client</li>
    <li>python38Packages.bellows</li>
    <li>python38Packages.pyhs100</li>
    <li>python38Packages.voluptuous</li>
    <li>python38Packages.voluptuous-serialize</li>
    <li>python38Packages.yeelight</li>
    <li>python38Packages.zha-quirks</li>
    <li>python38Packages.zigpy</li>
    <li>python38Packages.zigpy-cc</li>
    <li>python38Packages.zigpy-deconz</li>
    <li>python38Packages.zigpy-xbee</li>
    <li>python38Packages.zigpy-zigate</li>
  </ul>
</details>

**Pull request merged**